cli/lib/nf3.mjs

```diff
@@ -186,7 +196,7 @@ class Nf3 {
     let tx;
     await this.nonceMutex.runExclusive(async () => {
       // if we don't have a nonce, we must get one from the ethereum client
-      if (!this.nonce) this.nonce = await this.web3.eth.getTransactionCount(this.ethereumAddress);
+      this.nonce = await this.web3.eth.getTransactionCount(this.ethereumAddress);
```

Comment:
If we do this, then we are not managing nonce increments ourselves. We used to think that web3 had trouble updating its nonces so we managed them locally, but given this is working, we were probably wrong and simply using a Mutex was sufficient.  Thus I think this code change is fine but then we should remove comments ln 198, 193

Reply:
web3 seems to work just fine managing nonces, at least I had 0 problems removing the local nonce management. Actually, some tests were failing exactly because of nonce conflicts. I'll just remove comments in 198, 193

Reply:
Yes, I think our current tests would expose any issues here so I'm good with this change.